keras_contrib/applications/densenet.py

```diff
@@ -323,3 +467,146 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
+        dropout_rate: dropout rate
+        weight_decay: weight decay
+        nb_layers_per_block: number of layers in each dense block.
+            Can be a positive integer or a list.
+            If positive integer, a set number of layers per dense block.
+            If list, nb_layer is used as provided. Note that list size must
+            be (nb_dense_block + 1)
+        nb_upsampling_conv: number of convolutional layers in upsampling via subpixel convolution
+        upsampling_type: Can be one of 'upsampling', 'deconv', 'atrous' and
+            'subpixel'. Defines type of upsampling algorithm used.
+        batchsize: Fixed batch size. This is a temporary requirement for
+            computation of output shape in the case of Deconvolution2D layers.
+            Parameter will be removed in next iteration of Keras, which infers
+            output shape of deconvolution layers automatically.
+        input_shape: Only used for shape inference in fully convolutional networks.
+
+    Returns: keras tensor with nb_layers of conv_block appended
+    '''
+
+    concat_axis = 1 if K.image_dim_ordering() == "th" else -1
+
+    if concat_axis == 1:  # th dim ordering
+        _, rows, cols = input_shape
+    else:
+        rows, cols, _ = input_shape
+
+    if reduction != 0.0:
+        assert reduction <= 1.0 and reduction > 0.0, "reduction value must lie between 0.0 and 1.0"
+
+    # check if upsampling_conv has minimum number of filters
+    # minimum is set to 12, as at least 3 color channels are needed for correct upsampling
+    assert nb_upsampling_conv > 12 and nb_upsampling_conv % 4 == 0, "Parameter `upsampling_conv` number of channels must " \
+                                                                    "be a positive number divisible by 4 and greater " \
+                                                                    "than 12"
+
+    # layers in each dense block
+    if type(nb_layers_per_block) is list or type(nb_layers_per_block) is tuple:
+        nb_layers = list(nb_layers_per_block)  # Convert tuple to list
+
+        assert len(nb_layers) == (nb_dense_block + 1), "If list, nb_layer is used as provided. " \
+                                                       "Note that list size must be (nb_dense_block + 1)"
+
+        final_nb_layer = nb_layers[-1]
+        nb_layers = nb_layers[:-1]
+
+    else:
+        final_nb_layer = nb_layers_per_block
+        nb_layers = [nb_layers_per_block] * nb_dense_block
+
+    if bottleneck:
+        nb_layers = [int(layer // 2) for layer in nb_layers]
+
+    # compute initial nb_filter if -1, else accept users initial nb_filter
+    if nb_filter <= 0:
+        nb_filter = 2 * growth_rate
+
+    # compute compression factor
+    compression = 1.0 - reduction
+
+    # Initial convolution
+    x = Convolution2D(48, 3, 3, init="he_uniform", border_mode="same", name="initial_conv2D", bias=False,
```

Comment:
This is the initial convolution kernel. As suggested in the Tiramisu paper, it is always 48. Using the variable over here means the user can alter the initial number of convolutions.